src/v/cluster/tm_stm.cc

```diff
@@ -288,11 +305,14 @@ tm_stm::do_update_tx(tm_transaction tx, model::term_id term) {
 ss::future<checked<tm_transaction, tm_stm::op_status>>
 tm_stm::mark_tx_preparing(
   model::term_id expected_term, kafka::transactional_id tx_id) {
-    auto ptx = _mem_txes.find(tx_id);
-    if (ptx == _mem_txes.end()) {
+    auto ptx = get_tx(tx_id);
```

Comment:
please see my other comment about returning `checked` from get_tx, we don't need to modify `_mem_txes.find` at all! and actually it's better not modify it because it's already tested and it uses an implicit safety check: `_mem_txes` contains only transactions started in this term.